src/framework/server.ts

```diff
@@ -76,8 +61,46 @@ export type SettingsInput = createExpressGraphql.OptionsData &
     plugins: Plugin.RuntimeContributions[]
     contextContributors: ContextContributor<any>[]
   }
+/**
+ * [API Reference](https://nexus-future.now.sh/#/references/api?id=server)  ⌁  [Guide](todo)
+ *
+ * ### todo
+ *
+ */
+export interface Server {
+  /**
+   * todo
+   */
+  start: () => Promise<void>
+  /**
+   * todo
+   */
+  stop: () => Promise<void>
+}
+
+interface ServerWithInstance<T extends any> extends Server {
+  instance: T
+}
+
+interface CustomServerHookInput {
+  schema: GraphQL.GraphQLSchema
+  defaultServer: ServerWithInstance<Express>
```

Comment:
This was indeed the original goal. But your comment made me realize that nothing was ready for plugins to override the type anyway so it might be a bit premature. I reverted and "hardcoded" express into its own type for now!

Regarding `original` vs `default`, I think `original` is probably better. That's usually how people name their variable when overriding functions. FWIW, I have an example in mind, taken from the TS compiler API documentation

![image](https://user-images.githubusercontent.com/25233379/74232317-d3df0500-4cc8-11ea-96b3-7e1ec4a37d56.png)

Notice how both variables are called `orig<something>`, where `orig` is most likely referring to `original`.

TLDR: I renamed to property to `originalServer`